plugins/redis/redis.cpp

```diff
@@ -201,7 +201,7 @@ protected :
     void assertOnErrorWithCmdMsg(const redisReply * reply, const char * cmd, const char * key = nullptr);
     void assertConnection(const char * _msg);
     void assertConnectionWithCmdMsg(const char * cmd, const char * key = nullptr);
-    void fail(const char * cmd, const char * errmsg, const char * key = nullptr);
+    [[noreturn]] void fail(const char * cmd, const char * errmsg, const char * key = nullptr);
```

Comment:
I'm not sure this syntax is supported by all compilers (especially msvc).

Looking at jexcept.hpp we use:

```
__declspec(noreturn) void jlib_decl throwStringExceptionV(int code, const char *format, ...) __attribute__((noreturn));
```